packages/freesewing.shared/components/workbench/inputs/measurement.js

```diff
@@ -59,20 +58,19 @@ const MeasurementInput = ({ m, gist, app, updateMeasurements }) => {
           type="text"
           placeholder={title}
           className={`
-            input input-lg input-bordered grow text-base-content
-            ${valid === false && 'input-error'}
-            ${valid === true && 'input-success'}
-            border-r-0
+            input input-lg input-bordered grow text-base-content border-r-0
+            ${isValid() === false && 'input-error'}
+            ${isValid() === true && 'input-success'}
           `}
           value={val}
           onChange={update}
         />
         <span role="img" className={`bg-transparent border-y
-          ${valid === false && 'border-error text-neutral-content'}
-          ${valid === true && 'border-success text-neutral-content'}
-          ${valid === '' && 'border-base-200 text-base-content'}
+          ${isValid() === false && 'border-error text-neutral-content'}
+          ${isValid() === true && 'border-success text-neutral-content'}
+          ${isValid() === null && 'border-base-200 text-base-content'}
        `}>
-          {valid === ''
+          {isValid() === null
```

Comment:
the lines below look odd. as is does the conditional twice

Reply:
It's a chained ternary.
is it null
yes => ''
no => is it then perhaps valid
yes => thumbs up
no => smiley

Which is indeed a bit terse, I'll rewrite it

Reply:
ok, I rewrote this